src/commands/command.interface.ts

```diff
@@ -0,0 +1,8 @@
+import { Client } from 'discord.js';
+
+export interface ICommand {
+    name: string;
```

Comment:
As for 1) I'd suggest changing helpMessage - as you suggested to a validation function and description to the actual help message. Thus when it doesn't pass `helpMessage / validate (suggestion)` it will call `message.channel.send(description)`. Maybe append something like. `Syntax Error:\n ${description}`.  
As for 2) alright - sure, I don't mind working with a singleton! Seems more effective to me ^^!

Reply:
I'm not sure I understand what you're saying in response to (1). Could you provide a bit of code example? Or just make the change and I'll have a look then.

Reply:
Changed the code accordingly. You can review how I did it 👍